appservice/src/SiteWrapper.ts

```diff
@@ -170,6 +176,61 @@ export class SiteWrapper {
+        const oAuth2Token: string = (await client.listSourceControls())[0].token;
+        if (!oAuth2Token) {
+            this.showGitHubAuthPrompt(node);
+            return;
+        }
+
+        const gitHubUser: Object[] = await this.getJsonRequest('https://api.github.com/user', oAuth2Token, node);
+
+        const gitHubOrgs: Object[] = await this.getJsonRequest('https://api.github.com/user/orgs', oAuth2Token, node);
+        const orgQuickPicks: IQuickPickItemWithData<{}>[] = this.createQuickPickFromJsons([gitHubUser], 'login', undefined, ['repos_url']).concat(this.createQuickPickFromJsons(gitHubOrgs, 'login', undefined, ['repos_url']));
+        const orgQuickPick: gitHubOrgData = (await vscode.window.showQuickPick(orgQuickPicks, { placeHolder: 'Choose your organization.', ignoreFocusOut: true })).data;
+
+        const gitHubRepos: Object[] = await this.getJsonRequest(orgQuickPick.repos_url, oAuth2Token, node);
+        const repoQuickPicks: IQuickPickItemWithData<{}>[] = this.createQuickPickFromJsons(gitHubRepos, 'name', undefined, ['url', 'html_url']);
+        const repoQuickPick: gitHubReposData = (await vscode.window.showQuickPick(repoQuickPicks, { placeHolder: 'Choose project.', ignoreFocusOut: true })).data;
+
+        const gitHubBranches: Object[] = await this.getJsonRequest(`${repoQuickPick.url}/branches`, oAuth2Token, node);
+        const branchQuickPicks: IQuickPickItemWithData<{}>[] = this.createQuickPickFromJsons(gitHubBranches, 'name');
+        const branchQuickPick: IQuickPickItemWithData<{}> = await vscode.window.showQuickPick(branchQuickPicks, { placeHolder: 'Choose branch.', ignoreFocusOut: true });
+
+        const siteSourceControl: SiteSourceControl = {
+            location: this.location,
+            repoUrl: repoQuickPick.html_url,
+            branch: branchQuickPick.label,
+            isManualIntegration: false,
+            deploymentRollbackEnabled: true,
+            isMercurial: false
+        };
+
+        this.log(outputChannel, `"${this.appName}" is being connected to GitHub repo. This may take a several minutes...`);
+        try {
+            await client.webApps.createOrUpdateSourceControlWithHttpOperationResponse(this.resourceGroup, this.name, siteSourceControl);
+        } catch (err) {
+            try {
+                // a resync will fix the first broken build
+                // https://github.com/projectkudu/kudu/issues/2277
+                await client.webApps.syncRepository(this.resourceGroup, this.name);
+            } catch (error) {
+                const parsedError: IParsedError = JSON.parse(parseError(error).message);
+                // The portal returns 200, but is expecting a 204 which causes it to throw an error even after a successful sync
+                if (parsedError.statusCode !== 200) {
+                    throw error;
+                }
+            }
+        }
+
+        this.log(outputChannel, `"${this.appName}" has been connected to GitHub repo.`);
```

Comment:
Is this message redundant? Considering you already have a message at the end of editScmType